app/src/main/java/com/engineering/dokkan/view/Favourite/Favourite_Fragment.java

```diff
@@ -0,0 +1,72 @@
+package com.engineering.dokkan.view.Favourite;
+
+import android.view.View;
+
+
+import androidx.navigation.NavController;
+import androidx.navigation.NavOptions;
+import androidx.navigation.Navigation;
+import androidx.viewpager.widget.ViewPager;
+
+import com.engineering.dokkan.R;
+import com.engineering.dokkan.view.base.BaseFragment;
+import com.google.android.material.tabs.TabLayout;
+
+public class Favourite_Fragment extends BaseFragment{
+
+    private TabLayout tabLayout;
+    private ViewPager viewPager;
+    private ViewPagerAdapter viewadapter;
+
+
+
+    @Override
+    public int getLayoutId() {
+        return R.layout.fragment_favourite;
+    }
+
+
+    NavController getNavController() {
+        return Navigation.findNavController(getActivity(), R.id.nav_host_fragment);}
+
+
+    @Override
+    public void initializeViews(View view) {
+        tabLayout = view.findViewById(R.id.tablayout);
+        viewPager = view.findViewById(R.id.viewpager);
+        viewadapter = new ViewPagerAdapter(getActivity().getSupportFragmentManager());
+        viewadapter.AddFragment(new Item_Fragment(), "Item");
+        viewadapter.AddFragment(new Shop_Fragment(), "Shop");
+        viewPager.setAdapter(viewadapter);
+        tabLayout.setupWithViewPager(viewPager);
+        
+        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tabLayout) {
+                viewPager.setCurrentItem(tabLayout.getPosition());
+                switch (tabLayout.getPosition()) {
+                    case 0:
+                        getNavController().navigate(R.id.action_favourite_Fragment_to_item_Fragment);
+                        break;
+                    case 1:
+                        getNavController().navigate(R.id.action_favourite_Fragment_to_shop_Fragment);
+                        break;
+                }
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tabLayout) {
+
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tabLayout) {
```

Comment:
navigating on tap selection make the movement to another fragment which not having the tabs and pager
viewpager is responsible for navigating betweeen shop fragment and item fragment 
so delet this tabLayout.addOnTabSelectedListener{}